proxy/sources/mangadventure.py

```diff
@@ -0,0 +1,156 @@
+        groups = {
+            str(key): value for key, value in enumerate({
+                group for chapter in data["chapters"].values()
+                for group in chapter["groups"].keys()
+            })
+        }
+        chapters = {}
+        for num, chapter in data["chapters"].items():
+            chapters[num] = {
+                "title": chapter["title"],
+                "volume": chapter["volume"]
+            }
+            group = next(
+                k for k in groups.keys() for g in
+                chapter["groups"].keys() if g == groups[k]
+            )
+            chapters[num]["groups"] = {
+                group: self.wrap_chapter_meta(encode(
+                    base + chapter["volume"] + "/" + num
+                ))
+            }
+            chapters[num]["release_date"] = {
+                group: int(chapter["last_updated"])
+            }
+
+        return SeriesAPI(
+            slug=meta_id,
+            title=data["title"],
+            description=data["description"],
+            author=data["author"],
+            artist=data["artist"],
+            groups=groups,
+            cover=data["cover"],
+            chapters=chapters,
+            series_name=data["title"]
+        )
+
+    @api_cache(prefix="ma_series_page_dt", time=3600)
+    def chapter_api_handler(self, meta_id: str):
+        scheme, domain, slug, vol, num = decode(meta_id).split("/", 4)
+        qs = f"?series={slug}&volume={vol}&number={num}"
+        url = f"{scheme}://{domain}/api/v2/pages{qs}&track=true"
+        res = get_wrapper(url, headers=self.headers)
+        if res.status_code != 200:
+            return None
+        pages = [page["image"] for page in loads(res.text)["results"]]
+        return ChapterAPI(series=slug, pages=pages, chapter=num)
+
+    @api_cache(prefix="ma_series_page_dt", time=600)
+    def series_page_handler(self, meta_id: str):
+        scheme, domain, slug = decode(meta_id).split("/", 2)
+        url = f"{scheme}://{domain}/api/v2/cubari/{slug}"
+        res = get_wrapper(url, headers=self.headers)
+        if res.status_code != 200:
+            return None
+        data = loads(res.text)
+
+        # simplified version of the gist mapper
+        chapters = [[
+            num, num,
+            chapter["title"],
+            num.replace(".", "-"),
+            list(chapter["groups"].keys())[0],
+            self.parse_date(chapter["last_updated"]),
+            chapter["volume"],
+        ] for num, chapter in data["chapters"].items()][::-1]
+
+        return SeriesPage(
+            series=data["title"],
+            alt_titles=data["alt_titles"],
+            alt_titles_str=None,
+            slug=meta_id,
+            cover_vol_url=data["cover"],
+            metadata=data["metadata"],
+            synopsis=data["description"],
+            author=data["author"],
+            chapter_list=chapters,
+            original_url=data["original_url"],
```

Comment:
This needs the domain as well since it looks like the API returns an absolute path.